Examples article on temporary ratings (#203)
trolie/spec: docs/example-narratives/temporary-ratings.md

Reviewer: I think this should have a section discussing how these temporary ratings appear in the snapshots. I'll get this done as soon as I can as it will affect that section.
https://github.com/trolie/spec/issues/200

---
title: Temporary Ratings
parent: Usage Examples
nav_order: 7
toc: true
---

# Temporary Ratings

> FERC order 881 includes a provision for a "reliability safety valve" that allows grid operators

Reviewer: ```suggestion
FERC Order 881 includes a provision that allows grid operators
```

to temporarily set aside AARs to address immediate threats to reliability by 

> temporarily using an alternate static rating for a facility.  These temporary ratings used 

Reviewer: ```suggestion
[temporarily using an alternate static rating](https://www.federalregister.gov/d/2021-27735/p-71) for a facility.  These temporary ratings used 
```

must be documented and stored in a database, accessible for 5 years.  

TROLIE includes two kinds of temporary ratings objects:

> * **Temporary AAR Exceptions** are exceptions to the ratings methodology on facilities using AARs. These are records that include a time-bound set of static ratings against a particular resource, with a free-text explanation as to why the exception occurred.  This typically relates to an outage or other physical need to "de-rate" a line.  

Reviewer: ```suggestion
* [**Temporary AAR Exceptions**](../concepts##temporary-aar-exception) are temporary exemptions from AAR obligations due to exceptional circumstances. These are records that include a time-bound set of static ratings against a particular resource, with a free-text explanation as to why the exception occurred.  This typically relates to an outage or other physical need to "de-rate" a line.  
```

> * **Seasonal Overrides** are used for updating seasonal ratings out-of-band of a Transmission Provider's typical seasonal rating update schedule and process.  Their structure is identical to Temporary AAR Exceptions.  However, the use cases are quite different, and generally only apply to certain corner cases.  Examples include:

Reviewer: ```suggestion
* [**Seasonal Overrides**](../concepts#seasonal-overrides) are to be used by the Clearinghouse in lieu of seasonal ratings on a temporary basis. Their structure is identical to Temporary AAR Exceptions (assuming DAY/NIGHT seasonal ratings are not involved).  However, the use cases are quite different.  Examples include:
```

>   * For resources that are AAR-exempt and only support seasonal ratings, the seasonal override serves a similar purpose to Temporary AAR Exceptions.  

Reviewer: ```suggestion
  * For resources that are AAR-exempt and only support seasonal ratings, the Seasonal Override serves a similar purpose to Temporary AAR Exceptions.  
```

>   * Resources at Transmission Provider seams may have seasonal ratings updated at offset schedules.  Seasonal Overrides provide a tool to handle these offsets in operational systems.  

Reviewer: ```suggestion
  * Adjacent Transmission Providers update seasonal ratings independently. Sometimes these changes are temporary or occur on an offset schedule. Seasonal Overrides provide a tool to handle these updates without requiring a model build. 
```

>   * If for any reason it makes sense to temporarily update the recourse rating for a given resource.  

Reviewer: ```suggestion
  * Where the seasonal rating is used as a recourse rating, a Seasonal Override is effectively the recourse rating.
```

```json
{% include_relative examples/temporary-aar-exception-get.json %}
```

## Updating and Deleting Temporary Ratings
Once created, the temporary ratings may be updated.  This is often needed if the 
anticipated end time changes, or if more detail needs to be added to the
reason.  For the Temporary AAR Exception created above, this may be achieved by 
invoking [updateTemporaryAARException](../spec#tag/Temporary%20AAR%20Exceptions/operation/updateTemporaryAARException).  

Assuming the `output.json` file above was modified as desired, the Temporary
AAR Exception may be updated with the following command.  Note that the ID
returned from the create operation must be used in the URL:

```bash
curl -d @output.json \
-X PUT \
-H "Content-Type: application/vnd.trolie.temporary-aar-exception.v1+json" \
-H "Accept: application/vnd.trolie.temporary-aar-exception.v1+json" \
"https://trolie.example.com/temporary-aar-exceptions/46f7212b-1633-4c30-ba71-c6e987b2ded7"
```

Finally, the temporary rating may also have been created by mistake.  It may be 
canceled by deleting it using 
[deleteTemporaryAARException](../spec#tag/Temporary%20AAR%20Exceptions/operation/deleteTemporaryAARException)

> This can be done with the following command:

Reviewer: One caveat here--it can't be canceled per se, if it is already underway. "Cancel" in that case would simply updated the end datetime to now.

Author: Added a little clarification to this point.

---
title: Temporary Ratings
parent: Usage Examples
nav_order: 7
toc: true
---

# Temporary Ratings
FERC Order 881 includes a provision that allows grid operators
to temporarily set aside AARs to address immediate threats to reliability by 
[temporarily using an alternate static rating](https://www.federalregister.gov/d/2021-27735/p-71) for a facility.  These temporary ratings used 
must be documented and stored in a database, accessible for 5 years.  

TROLIE includes two kinds of temporary ratings objects:

> * [**Temporary AAR Exceptions**](../concepts##temporary-aar-exception) are temporary exemptions from AAR obligations due to exceptional circumstances. These are records that include a time-bound set of static ratings against a particular resource, with a free-text explanation as to why the exception occurred.  This typically relates to an outage or other physical need to "de-rate" a line.  

Reviewer: ```suggestion
* [**Temporary AAR Exceptions**](../concepts#temporary-aar-exception) are temporary exemptions from AAR obligations due to exceptional circumstances. These are records that include a time-bound set of static ratings against a particular resource, with a free-text explanation as to why the exception occurred.  This typically relates to an outage or other physical need to "de-rate" a line.  
```
